compiler/ilgen/IlBuilder.cpp

```diff
@@ -1564,6 +1564,114 @@ IlBuilder::AtomicAdd(TR::IlValue * baseAddress, TR::IlValue * value)
    return AtomicAddWithOffset(baseAddress, NULL, value);    
    }
 
+/**
+ * \brief
+ *  the service is for generating a treetop for transaction begin when the user needs to use transactional memory.
+ *
+ * lock (TransactionBegin)
+ *    ...
+ *    transactionalRegion
+ *    ...
+ * unlock (TrasanctionEnd) 
```

Comment:
While it is possible to implement transactional lock elision using this service, it doesn't directly implement it; IMO, it's misleading to talk about a lock in this context here. I'd drop this section, and replace with a pseudocode implementation or basic-block diagram of control flow.